staging/src/k8s.io/kubectl/pkg/metricsutil/metrics_printer.go

```diff
@@ -200,49 +202,70 @@ func (printer *TopCmdPrinter) PrintPodMetrics(metrics []metricsapi.PodMetrics, p
 		if withNamespace {
 			printValue(w, NamespaceColumn)
 		}
+		if len(pods.Items) > 0 {
```

Comment:
```suggestion
		if pods != nil && len(pods.Items) > 0 {
```

Otherwise if the pointer is nil it might get a panic :)

Reply:
would this be possible, since this has to be passed into the function to begin with? if pods is nil, that would mean someone passed in nil directly here correct? because they will get an error if they dont pass pods in.

Also if we do this, should we also add something inside of the createPodMap function?

```
func createPodMap(pods *v1.PodList) (podMap map[string]v1.Pod) {
	podMap = make(map[string]v1.Pod)
	for _, pod := range pods.Items {
		podMap[pod.Name] = pod
	}
	return podMap
}
```

That way we dont have to check explicity for pods when calling this function on line 216.
```
podMap := createPodMap(pods)
```

The thought is, we always call this so we dont have to check if pods exist or not. We only need to check if individual pods exist inside the list.

The default state of pods is ```var pods *v1.PodList = new(v1.PodList)``` so if we get an error when getting pods, we return error and this function fails to begin with. we don't continue.

Reply:
Trying to keep the code a little clean and not have to check stuff all over the place. This is a default param, so not sure how it would ever be ok for someone to pass nil as the podsList interface is what is expected.